### PR TITLE
Experimental chat lag fix: Send all messages in a single network round trip if there are a lot of messages.

### PR DESCRIPTION
**Pull request by BenLubar**
[PERF]

## About the PR

Based on the fact that I don't experience chat lag at all on localhost, I believe the lag caused by large quantities of chat updates is due to network round trips. This PR adds a queue that activates after a certain number of chat updates are performed in a single tick, which then prevents chat updates from being sent for a set period of time, after which one chat update is sent containing all the pending data.

### CODE REVIEW

**Comment by BenLubar** [comment]
It works in the sense that it doesn't make the server or client explode, but I haven't been able to test it with actual latency (or fake latency, for that matter; I need to look into clumsy.exe).

File: code/datums/browserOutput.dm [hunk 1]
```diff
@@ -392,6 +400,24 @@ var/global
 			//Client sucks at loading things, put their messages in a queue
 			C.chatOutput.messageQueue["[C.chatOutput.messageQueue.len]"] = list("message" = message, "group" = group)
 		else
+			if (C?.chatOutput)
+				var/T = TIME
+				if (islist(C.chatOutput.burstQueue))
+					C.chatOutput.burstQueue["[C.chatOutput.burstQueue.len]"] = list("message" = message, "group" = group)
```
Comment (ZeWaka) [inline code/datums/browserOutput.dm:406]: length

File: code/datums/browserOutput.dm [hunk 2]
```diff
@@ -392,6 +400,24 @@ var/global
 			//Client sucks at loading things, put their messages in a queue
 			C.chatOutput.messageQueue["[C.chatOutput.messageQueue.len]"] = list("message" = message, "group" = group)
 		else
+			if (C?.chatOutput)
+				var/T = TIME
```
Comment (ZeWaka) [inline code/datums/browserOutput.dm:404]: var name

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Comment by ZeWaka** [comment]
Tested on develop, works.

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**